### ISSUE TITLE
Daylight Savings cutting off last ocurrence.

### ISSUE THREAD

**Issue by wilferraciolli** [ical4j/ical4j#99]
Recurrence Rule fails to get last instance for UTC date range. I am writing a program to get candidate date and time within 6 months, when the time goes back/forward by 1 hours I either miss or get 1 extra occurrence. 

`Recur recur = new Recur("FREQ=WEEKLY;INTERVAL=1;BYDAY=FR;WKST=MO;UNTIL=20170127T003000Z;");

```
    // Define the start and end dates
    DateTime startDate = new DateTime("20160727T0030000Z");
    startDate.setUtc(true);

    DateTime endDate = (DateTime) recur.getUntil();
    endDate.setUtc(true);

   / /
    DateList dateList = recur.getDates(startDate, endDate, Value.DATE_TIME);

    // Loop through the events
    for (int x = 0; x < dateList.size(); x++) {
        String d1 = dateList.get(x).toString();
        System.out.println(d1);
    }`
```

The code above return instances up to 20/01/2017 but it should be 27/01/2017

If I change until property of the recurrence rule to be at the end of the day it works 
Eg:        Recur recur = new Recur("FREQ=WEEKLY;INTERVAL=1;BYDAY=FR;WKST=MO;UNTIL=20170127T235959Z

**Comment by trippytango** [comment]
We had a bug related to this a couple of months ago, we thought it might of been the library but we just decided to let it pass. Would like this to be fixed as we've had a couple of customers question why this isn't working. +1

**Comment by artheus** [comment]
This kind of problem is actually a very common one when it comes to recurring events. I have noticed this kind of behaviour when I have written and worked on libraries using DateTime earlier.
The hard part about this is that you are using DateTime, where Time is as important as Date.

But as you say, this is a bug. And the correct behaviour is documented in the RFC 5545 document ([Read it here, link to correct section of the document.](https://tools.ietf.org/html/rfc5545#page-41))

> The UNTIL rule part defines a DATE or DATE-TIME value that bounds
      the recurrence rule in an inclusive manner.  If the value
      specified by UNTIL is synchronized with the specified recurrence,
      this DATE or DATE-TIME becomes the last instance of the
      recurrence.

**Comment by artheus** [comment]
As mentioned in the event stream here I added a pull request for this.

The unit test added is the exact example from @wilferraciolli issue message.

**Comment by wilferraciolli** [comment]
Awesome

**Comment by benfortuna** [comment]
The fix has been applied as part of recent refactoring efforts. Will be released in v3.0.4
